TracyOpenGL.hpp

```diff
@@ -173,6 +167,38 @@ class GpuCtx
     }
 
 private:
+    void SyncClockAndSendContextInfo()
+    {
+        ZoneScopedC( Color::Red4 );
+
+        int64_t tgpu;
+        glGetInteger64v( GL_TIMESTAMP, &tgpu );
+        int64_t tcpu = Profiler::GetTime();
+
+        GLint bits;
+        glGetQueryiv( GL_TIMESTAMP, GL_QUERY_COUNTER_BITS, &bits );
+
+        const float period = 1.f;
+        const auto thread = GetThreadHandle();
+
+        TracyLfqPrepare(QueueType::GpuNewContext);
+        MemWrite( &item->gpuNewContext.cpuTime, tcpu );
+        MemWrite( &item->gpuNewContext.gpuTime, tgpu );
+        MemWrite( &item->gpuNewContext.thread, thread );
+        MemWrite( &item->gpuNewContext.period, period );
+        MemWrite( &item->gpuNewContext.context, m_context );
+        MemWrite( &item->gpuNewContext.flags, uint8_t(0) );
+        MemWrite( &item->gpuNewContext.type, GpuContextType::OpenGl );
+
+#if defined( TRACY_ON_DEMAND )
+        GetProfiler().DeferItem( *item );
+#elif defined( TRACY_ON_DEMAND_GPU_SYNC )
```

Comment:
Is the condition order here the way you intended to be? `TRACY_ON_DEMAND_GPU_SYNC` does not make sense without `TRACY_ON_DEMAND`.

Reply:
Whoops sorry, forgot to repush the ammended commit before opening the PR.

Reply:
Do you prefer that I nest the `#ifdefs` ? or just in the correct order ?

Reply:
I think that nesting is better at expressing the correct usage pattern.

Reply:
I updated all the checks to also check `TRACY_ON_DEMAND`.
The other option would have been to `#undef TRACY_ON_DEMAND_GPU_SYNC` if `TRACY_ON_DEMAND` was not defined.